plugin-server/src/cdp/cdp-consumers.ts

```diff
@@ -612,6 +605,8 @@ export class CdpOverflowConsumer extends CdpConsumerBase {
         return await runInstrumentedFunction({
             statsKey: `cdpConsumer.handleEachBatch.executeOverflowedFunctions`,
             func: async () => {
+                await this.groupsManager.enrichGroups(invocationGlobals.map((x) => x.globals))
```

Comment:
Might be worth adding it here as well.

```suggestion
                // TODO: Add a helper to hog functions to determine if they require groups or not and then only load those
                await this.groupsManager.enrichGroups(invocationGlobals.map((x) => x.globals))
```